cmd/docker/docker.go

```diff
@@ -30,7 +30,7 @@ import (
 
 func main() {
 	err := dockerMain(context.Background())
-	if err != nil && !errdefs.IsCancelled(err) {
+	if err != nil && !errdefs.IsCancelled(err) && !errors.Is(err, context.Canceled) {
```

Comment:
Curious; errdefs.IsCancelled doesn't detect these? (that sounds like a possible bug if it doesn't)

Reply:
It doesn't, no. It checks that it implements the `errdefs` cancelled interface, not for `context.Canceled`.
```go
// ErrCancelled signals that the action was cancelled.
type ErrCancelled interface {
	Cancelled()
}
```

Reply:
But don't we want to be specific to only ignoring the _user cancelled the process_ vs _something/someone cancelled the process_? AFAIK `context.Canceled` can be a multitude of things and not just user cancelled. 

See https://github.com/docker/cli/pull/5760

Reply:
What else could cancel it though?

Reply:
Anything using `context.WithCancel()` can call the `cancel` func inside its own logic and return `ctx.Err()`. Since the error is ambiguous and can be used by anything down the call stack I'd prefer to impose a check on an error we know of. To me it seems it would be better to have something print out `context canceled` in a case where we don't properly handle it than every occurrence of `context.Canceled` silenced

Reply:
I'm okay with cherry-picking some of https://github.com/docker/cli/pull/5760 and using that to filter. Does that sound good @vvoland @thaJeztah?

Reply:
To be honest, I'd be inclined to consider function accepting `ctx` and returning `context.Canceled` when the input `ctx.Err() == nil` as buggy.

A logic using `context.WithCancel` should be able to handle that in its own boundary.

Reply:
To be honest, I'd be inclined to consider function accepting `ctx` and returning `context.Canceled` when the input `ctx.Err() == nil` as buggy.

A logic using `context.WithCancel` should be able to handle that in its own boundary.

Reply:
It is fine for a function to handle its own login with `context.WithCancel`. I'm just saying when it returns an error and we check it all the way at the entry point of the program, then it shouldn't be silenced automatically.